include/z64bgcheck.h

```diff
@@ -73,18 +71,39 @@ typedef struct {
     /* 0x10 */ s16 unk_10; // unused
 } BgCamFuncData; // size = 0x12
 
+// Macros for `WaterBox.properties`
+
+#define WATERBOX_BGCAM_INDEX_SHIFT 0
+#define WATERBOX_BGCAM_INDEX_MASK 0x000000FF
+#define WATERBOX_BGCAM_INDEX(properties) \
+    (((properties) >> WATERBOX_BGCAM_INDEX_SHIFT) & (WATERBOX_BGCAM_INDEX_MASK >> WATERBOX_BGCAM_INDEX_SHIFT))
+
+#define WATERBOX_LIGHT_INDEX_SHIFT 8
+#define WATERBOX_LIGHT_INDEX_MASK 0x00001F00
+#define WATERBOX_LIGHT_INDEX(properties) \
+    (((properties) >> WATERBOX_LIGHT_INDEX_SHIFT) & (WATERBOX_LIGHT_INDEX_MASK >> WATERBOX_LIGHT_INDEX_SHIFT))
+#define WATERBOX_LIGHT_INDEX_NONE 0x1F // warns and defaults to 0
+
+#define WATERBOX_ROOM_INDEX_SHIFT 13
+#define WATERBOX_ROOM_INDEX_MASK 0x0007E000
+#define WATERBOX_ROOM_INDEX(properties) \
+    (((properties) >> WATERBOX_ROOM_INDEX_SHIFT) & (WATERBOX_ROOM_INDEX_MASK >> WATERBOX_ROOM_INDEX_SHIFT))
+#define WATERBOX_ROOM_INDEX_ALL 0x3F // value for "room index" indicating "all rooms"
```

Comment:
room index? things like light index make sense to me cause it is used as an index to either light config or light settings, but this is just a room number isnt it?

Reply:
I didn't know room numbers were specifically called "room numbers" if that's what you mean, so change
`WATERBOX_ROOM_INDEX...` -> `WATERBOX_ROOM_NUMBER...` ?

Reply:
id even say just _ROOM is fine tbh. for example from the actor struct:
`/* 0x003 */ s8 room; // Room number the actor is in. -1 denotes that the actor won't despawn on a room change`

but yea people generally say "room number". the field is curRoom.num for example